modules/andes-core/broker/src/main/java/org/wso2/andes/amqp/QpidAndesBridge.java

```diff
@@ -488,7 +469,11 @@ private static void addLocalSubscriptionsForAllBindingsOfQueue(AMQQueue queue, S
                         Andes.getInstance().openLocalSubscription(subscriptionEvent);
                         alreadyAddedSubscriptions.add(subscriptionEvent);
                     }
+                } else {
+                    throw new AndesException("Binding already exists for the queue " + queue
+                            .getName());
```

Comment:
I don't think this is a limitation in Andes core, and it's better to have a clear separation between Andes core and transports. Everything before QpidBridge should be unaware of classes in Andes core and everything after QpidBridge should be unaware of transport specific classes. Only the bridge should be aware of classes in both sides. Therefore -1 in using AndesException.